script/tool/test/format_command_test.dart

```diff
@@ -0,0 +1,344 @@
+  });
+
+  List<String> _getAbsolutePaths(
+      Directory package, List<String> relativePaths) {
+    return relativePaths
+        .map((String path) => p.join(package.path, path))
+        .toList();
+  }
+
+  test('formats .dart files', () async {
+    const List<String> files = <String>[
+      'lib/a.dart',
+      'lib/src/b.dart',
+      'lib/src/c.dart',
+    ];
+    final Directory pluginDir = createFakePlugin(
+      'a_plugin',
+      packagesDir,
+      extraFiles: files,
+    );
+
+    await runCapturingPrint(runner, <String>['format']);
+
+    expect(
+        processRunner.recordedCalls,
+        orderedEquals(<ProcessCall>[
+          ProcessCall(
+              'flutter',
+              <String>['format', ..._getAbsolutePaths(pluginDir, files)],
+              packagesDir.path),
+        ]));
+  });
+
+  test('fails if flutter format fails', () async {
+    const List<String> files = <String>[
+      'lib/a.dart',
+      'lib/src/b.dart',
+      'lib/src/c.dart',
+    ];
+    createFakePlugin('a_plugin', packagesDir, extraFiles: files);
+
+    processRunner.mockProcessesForExecutable['flutter'] = <io.Process>[
+      MockProcess.failing()
+    ];
+    Error? commandError;
+    final List<String> output = await runCapturingPrint(
+        runner, <String>['format'], errorHandler: (Error e) {
+      commandError = e;
+    });
+
+    expect(commandError, isA<ToolExit>());
+    expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains('Failed to format Dart files: exit code 1.'),
+        ]));
+  });
+
+  test('formats .java files', () async {
+    const List<String> files = <String>[
+      'android/src/main/java/io/flutter/plugins/a_plugin/a.java',
+      'android/src/main/java/io/flutter/plugins/a_plugin/b.java',
+    ];
+    final Directory pluginDir = createFakePlugin(
+      'a_plugin',
+      packagesDir,
+      extraFiles: files,
+    );
+
+    await runCapturingPrint(runner, <String>['format']);
+
+    expect(
+        processRunner.recordedCalls,
+        orderedEquals(<ProcessCall>[
+          ProcessCall(
+              'java',
+              <String>[
+                '-jar',
+                javaFormatPath,
+                '--replace',
+                ..._getAbsolutePaths(pluginDir, files)
+              ],
+              packagesDir.path),
+        ]));
+  });
+
+  test('fails if Java formatter fails', () async {
+    const List<String> files = <String>[
+      'android/src/main/java/io/flutter/plugins/a_plugin/a.java',
+      'android/src/main/java/io/flutter/plugins/a_plugin/b.java',
+    ];
+    createFakePlugin('a_plugin', packagesDir, extraFiles: files);
+
+    processRunner.mockProcessesForExecutable['java'] = <io.Process>[
+      MockProcess.failing()
+    ];
+    Error? commandError;
+    final List<String> output = await runCapturingPrint(
+        runner, <String>['format'], errorHandler: (Error e) {
+      commandError = e;
+    });
+
+    expect(commandError, isA<ToolExit>());
```

Comment:
Why not also expect that the tool is using retcode 5? (`const int _exitJavaFormatFailed = 5;`)

Reply:
In practice we never actually use the exit code, so we've never bothered with testing them; I figured we could add tests later if we ever decide to do something with them. I use them in the code because it's easy, and makes the line self-documenting.